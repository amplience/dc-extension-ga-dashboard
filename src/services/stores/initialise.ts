import {
  contentItemFilter,
  editionFilter,
  slotFilter,
} from '../../stores/ga-query-filters';
import {
  breakdown,
  contentItemIdMapping,
  editionIdMapping,
  gaApiKey,
  gaClientEmail,
  gaClientId,
  gaViewId,
  slotIdMapping,
} from '../../stores/google-analytics';
import { currencyCode, locale } from '../../stores/localization';
import type { ExtensionInstallation } from '../extension-sdk/extension-sdk.service';

export function initialiseStores(config: ExtensionInstallation): void {
  gaViewId.set(config?.googleAnalyticsViewId || '');
  gaClientId.set(config?.googleAnalyticsClientId || '');
  gaClientEmail.set(config?.googleAnalyticsClientEmail);
  gaApiKey.set(config?.googleAnalyticsKey || '');

  contentItemIdMapping.set(config?.mappings?.contentItemId);
  editionIdMapping.set(config?.mappings?.editionId);
  slotIdMapping.set(config?.mappings?.slotId);

  contentItemFilter.set(config?.filters?.contentItemFilter);
  editionFilter.set(config?.filters?.editionFilter);
  slotFilter.set(config?.filters?.slotFilter);

  breakdown.set(
    config?.breakdown || {
      dimension: 'ga:deviceCategory',
      title: 'Device Breakdown',
    }
  );

  if (config?.localization?.locale) {
    locale.set(config?.localization?.locale);
  }

  if (config?.localization?.currencyCode) {
    currencyCode.set(config?.localization?.currencyCode);
  }
}
