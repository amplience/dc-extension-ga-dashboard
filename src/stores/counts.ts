import { writable, derived } from 'svelte/store';
import type {
  SearchIndexUsersCount,
  SearchIndexSearchesCount,
  SearchIndexNoResultsRate,
} from 'dc-management-sdk-js';

export interface Counts {
  usersCount?: SearchIndexUsersCount;
  searchesCount?: SearchIndexSearchesCount;
  noResultsRate?: SearchIndexNoResultsRate;
}

export const counts = writable<Counts>({});

export const searchesPerUser = derived(counts, ($counts) => {
  return $counts?.usersCount?.count && $counts?.searchesCount?.count
    ? Math.round(
        ($counts?.searchesCount?.count / $counts?.usersCount?.count) * 1e1
      ) / 1e1
    : 0;
});
