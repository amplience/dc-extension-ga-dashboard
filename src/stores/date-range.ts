import { writable } from './persisted-store';
import { formatDateAsISOString } from '../utils/date-format';

export const DAY = 24 * 60 * 60 * 1000;

export interface DateRange {
  from: string;
  to: string;
}

const NOW = new Date().setHours(0, 0, 0, 0).valueOf();

export const dateRange = writable<DateRange>('date-range', {
  from: formatDateAsISOString(new Date(NOW - DAY * 30)),
  to: formatDateAsISOString(new Date(NOW - DAY)),
});
