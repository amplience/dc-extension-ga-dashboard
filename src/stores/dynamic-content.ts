import type { DynamicContent, Hub } from 'dc-management-sdk-js';
import { writable } from 'svelte/store';

export const client = writable<DynamicContent>(null);

export const hub = writable<Hub>(null);
