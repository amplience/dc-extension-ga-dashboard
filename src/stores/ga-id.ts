import { writable as persistedWritable } from './persisted-store';

export const gaId = persistedWritable<string>('ga-id', null);
