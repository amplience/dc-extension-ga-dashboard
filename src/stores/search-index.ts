import { writable as persistedWritable } from './persisted-store';
import type { SearchIndex } from 'dc-management-sdk-js';
import { writable } from 'svelte/store';

export const selectedIndexId = persistedWritable<string>(
  'selectedIndexId',
  null
);
export const index = writable<SearchIndex>(undefined);
export const includeReplicas = persistedWritable<boolean>(
  'includeReplicas',
  true
);
