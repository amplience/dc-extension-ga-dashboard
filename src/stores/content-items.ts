import { get, writable } from 'svelte/store';
import { client } from './dynamic-content';

function createContentItems() {
  const { subscribe, set, update } = writable([]);

  return {
    subscribe,
    fetch: async (id) => {
      const cachedContentItem = get(contentItems).find(
        (item) => item.id === id
      );
      if (cachedContentItem) {
        return cachedContentItem;
      }
      const contentItem = await get(client).contentItems.get(id);
      update((n) => [...n, ...[contentItem]]);
      return contentItem;
    },
    reset: () => set([]),
  };
}

export const contentItems = createContentItems();
