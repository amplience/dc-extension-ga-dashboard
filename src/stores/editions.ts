import { get, writable } from 'svelte/store';
import { client } from './dynamic-content';

function createEditions() {
  const { subscribe, set, update } = writable([]);

  return {
    subscribe,
    fetch: async (id) => {
      const cachedEdition = get(editions).find((item) => item.id === id);
      if (cachedEdition) {
        return cachedEdition;
      }
      const edition = await get(client).editions.get(id);
      update((n) => [...n, ...[edition]]);
      return edition;
    },
    reset: () => set([]),
  };
}

export const editions = createEditions();
