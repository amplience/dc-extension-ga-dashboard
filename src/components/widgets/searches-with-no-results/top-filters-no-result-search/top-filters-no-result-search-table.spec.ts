import { tick } from 'svelte';
import { render } from '@testing-library/svelte';
import TopFiltersNoResultSearchTable from './top-filters-no-result-search-table.svelte';
import type { SearchIndex } from 'dc-management-sdk-js';
import getTopFiltersNoResultSearch from './services/top-filters-no-result-search.service';
import { index } from '../../../../stores/search-index';
import topFiltersNoResultSearch from './services/__fixtures__/top-filter-no-results-search';
import { formatDateAsISOString } from '../../../../utils/date-format';

jest.mock('./services/top-filters-no-result-search.service');

describe('TopFiltersNoResultSearchTable', () => {
  beforeEach(() => {
    index.set({ name: 'test-index-name' } as SearchIndex);
  });
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render the TopSearches component with no data', async () => {
    (getTopFiltersNoResultSearch as jest.Mock).mockReturnValue(
      Promise.resolve([])
    );
    const { container } = render(TopFiltersNoResultSearchTable, {
      search: 'q0',
      limit: 5,
    });

    await tick();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the TopSearches component with data', async () => {
    (getTopFiltersNoResultSearch as jest.Mock).mockReturnValue(
      Promise.resolve(topFiltersNoResultSearch)
    );
    const { container } = render(TopFiltersNoResultSearchTable, {
      search: 'q0',
      limit: 5,
    });

    await tick();
    expect(container.firstChild).toMatchSnapshot();
    expect(getTopFiltersNoResultSearch).toHaveBeenCalledWith(
      { name: 'test-index-name' },
      {
        endDate: formatDateAsISOString(subtractDays(1)),
        includeReplicas: true,
        limit: 5,
        search: 'q0',
        startDate: formatDateAsISOString(subtractDays(30)),
      }
    );
  });

  function subtractDays(days: number): Date {
    return new Date(new Date().setDate(new Date().getDate() - days));
  }
});
