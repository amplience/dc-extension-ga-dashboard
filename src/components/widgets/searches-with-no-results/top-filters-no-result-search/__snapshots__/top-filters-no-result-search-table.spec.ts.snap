// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopFiltersNoResultSearchTable should render the TopSearches component with data 1`] = `
<div>
  <div
    class="svelte-1mik51a"
  >
    <div
      class="data-table svelte-2e8um5"
    >
      <div
        class="data-table-slot secondary
    scrollable svelte-2e8um5"
      >
        <div
          class="data-table-head svelte-1nivie3"
        >
          <div
            class="data-table-row svelte-hige1l"
          >
            <div
              class="data-table-cell svelte-1x8v3rk"
              style="
  max-width:100%;
  justify-content:start;
  color:#999;"
            >
              Facets
               
              <div
                aria-expanded="false"
                class="svelte-usl6ua"
                data-testid="info-tip"
              >
                <div
                  class=" svelte-zpbvvv"
                  style="--icon-width: 24px; --icon-height: 24px;"
                >
                  <object
                    src="ic-asset-info.svg/"
                  />
                </div>
              </div>
               
            </div>
            <div
              class="data-table-cell svelte-1x8v3rk"
              style="
  max-width:25%;
  justify-content:flex-end;
  color:#999;"
            >
              Count
               
               
            </div>
            
          </div>
        </div>
         
        <div
          class="data-table-body"
        >
          <div
            class="data-table-row svelte-hige1l"
          >
            <div
              class="filter-values svelte-1mik51a"
            >
              <span
                class="filter-value svelte-1mik51a"
              >
                <span
                  class="filter-value-attribute svelte-1mik51a"
                >
                  brand
                </span>
                 
                <span>
                  :
                </span>
                 
                <span>
                  apple
                </span>
                 
              </span>
              <span
                class="filter-value svelte-1mik51a"
              >
                <span
                  class="filter-value-attribute svelte-1mik51a"
                >
                  price
                </span>
                 
                <span>
                  &gt;
                </span>
                 
                <span>
                  100
                </span>
                 
              </span>
            </div>
             
            <div
              class="data-table-cell svelte-1x8v3rk"
              style="
  max-width:25%;
  justify-content:flex-end;
  color:#333;"
            >
              3
            </div>
             
          </div>
          <div
            class="data-table-row svelte-hige1l"
          >
            <div
              class="filter-values svelte-1mik51a"
            >
              <span
                class="filter-value svelte-1mik51a"
              >
                <span
                  class="filter-value-attribute svelte-1mik51a"
                >
                  brand
                </span>
                 
                <span>
                  :
                </span>
                 
                <span>
                  apple
                </span>
                 
              </span>
            </div>
             
            <div
              class="data-table-cell svelte-1x8v3rk"
              style="
  max-width:25%;
  justify-content:flex-end;
  color:#333;"
            >
              1
            </div>
             
          </div>
          
        </div>
      </div>
    </div>
     
    
  </div>
</div>
`;

exports[`TopFiltersNoResultSearchTable should render the TopSearches component with no data 1`] = `
<div>
  <div
    class="svelte-1mik51a"
  >
    <div
      class="no-data-container svelte-1kq81j9"
    >
      <p>
        No facets for the selected date range
      </p>
    </div>
    
  </div>
</div>
`;
